src/sardana/taurus/qt/qtgui/extra_sardana/expdescription.py

```diff
@@ -321,6 +322,21 @@ def writeExperimentConfiguration(self, ask=True):
                 self.changeActiveMntGrp(mgname)
                 return False
 
+        # check path
+        path = self._localConfig['ScanDir']
+        if not os.access(path, os.F_OK):
+            try:
+                os.mkdir(path)
```

Comment:
On the one hand, using `os.mkdir` is not  a good option to check if the given path is writable or not since it only check one path level, I mean it will fail with subpath. e.g.  when you set up /foo/bar as path and neither foo and bar exist. On the other hand `spock` and `sardana` could be launched with different users which could have different permissions so you can not  advise the user about if sardana can write o no the data with this kind of checking.

Reply:
I understand.
Then it would be better to move the check to the door.

Reply:
Should the EnvironmentManager validate the values?
Which would be the right place to do the check?